torch/lib/THD/base/DataChannel.cpp

```diff
@@ -13,24 +13,24 @@
 
 namespace thd {
 
-DataChannel* DataChannel::newChannel(THDChannelType type) {
+DataChannel* DataChannel::newChannel(THDChannelType type, std::string init_method,
+                                     int world_size, std::string group_name) {
+  InitMethod::Config config = getInitConfig(init_method, world_size, group_name);
```

Comment:
I have just realized that this is bad placement for this initalization. When user wants to user `mpi` backend in PG mode it does not need to specify any initalization method. We should probably move it to particular `case`s.